apis/telemetry/v1alpha1/secret_refs.go

```diff
@@ -20,7 +20,7 @@ func (lp *LogPipeline) GetEnvSecretRefs() []SecretKeyRef {
 	var refs []SecretKeyRef
 
 	output := lp.Spec.Output
-	if output.IsHTTPDefined() {
+	if output.HTTP != nil {
```

Comment:
Why this change?

Reply:
Because the IsHTTPDefined() also checks that the Host field is defined (i.e. is a valid secretref), so then if you have an invalid Host secretref, you won't get the user and the password SRs either, and no error will be triggered, the fields will just be ignored and not appended to the returned SRs.